### ISSUE TITLE
Colo{,u}r Sampler does not update colours when position is updated.

### ISSUE THREAD

**Issue by DianaProbst** [AdobeDocs/uxp-photoshop#346]
Issue in /src/pages/ps_reference/classes/colorsampler.md

I have a colour sampler, cs, at position 1500, 100.  It was moved from its original position at 100, 100.  It's got a valid position which I can return.

![image](https://user-images.githubusercontent.com/6243444/199797903-a6b429a4-2c6a-4017-ac0f-d0f726aa5c4d.png)

However, the sampler has not updated, and does not update when I query it.

![image](https://user-images.githubusercontent.com/6243444/199798102-fa60bf95-3a63-47fb-bee9-bafd4eb26515.png)

It consistently picks up the colour from where it began.

**Comment by jardicc** [comment]
Yes... original value is cached internally and not updated properly. I will ask my friend to fix it.

**Comment by undavide** [comment]
What if you access the colorSampler by index?

**Comment by DianaProbst** [comment]
For colour read color throughout.

![image](https://user-images.githubusercontent.com/6243444/199945606-34ebed7f-034b-45d2-a737-1e6d80d2649c.png)
colourSampler by index seems to be giving me the correct number.  I also checked there was no index [1], and there's not.

An extra `cs = app.activeDocument.colorSamplers[0];` gets around the problem, at least.

**Comment by undavide** [comment]
Can I ask you what's the scenario that produces this disconnect? Usually when a script is in charge and then, for some reason, the control is given back to a user who might change the PS state, problems can arise. Getting the sampler from the collection via array notation triggers a full refresh instead.

**Comment by DianaProbst** [comment]
I'm trying to pick two colours for an APCA contrast check.  Everything I do is in a couple of modal functions.  I'm afraid I can't give you the code because I've been experimenting rather than writing a complete js file, so I've not got much beyond notes on what does and doesn't work, yet.  However, it's pretty much as above: there isn't any input from the user, just manipulation of the colour samplers.  I was creating one, taking the sample via the cs shortcut, moving it, and then taking another sample, but I've changed that so now I'm creating two in different places, and sampling from each via the index.  This seems to be avoiding all problems with caching, as you note.
